server/base/src/main/java/org/apache/accumulo/server/manager/state/AbstractTabletStateStore.java

```diff
@@ -38,16 +40,22 @@
 public abstract class AbstractTabletStateStore implements TabletStateStore {
 
   private final Ample ample;
+  private final Supplier<ServiceLock> lock;
 
-  protected AbstractTabletStateStore(ClientContext context) {
+  protected AbstractTabletStateStore(ClientContext context, Supplier<ServiceLock> lock) {
     this.ample = context.getAmple();
+    this.lock = lock;
+  }
+
+  protected ServiceLock getLock() {
+    return lock.get();
   }
 
   @Override
   public void setLocations(Collection<Assignment> assignments) throws DistributedStoreException {
     try (var tabletsMutator = ample.conditionallyMutateTablets()) {
       for (Assignment assignment : assignments) {
-        var conditionalMutator = tabletsMutator.mutateTablet(assignment.tablet)
+        var conditionalMutator = tabletsMutator.mutateTablet(assignment.tablet, lock.get())
```

Comment:
Looking at this, is it possible to move the passing of the lock to the construction of the mutator? It seems like it makes more sense there as the scope of the lock should be tied to the mutator instance life. This seems cleaner vs having to pass the lock to every mutateTablet() call.

For example:

```java
    try (var tabletsMutator = ample.conditionallyMutateTablets(lock.get())) {
      for (Assignment assignment : assignments) {
        var conditionalMutator = tabletsMutator.mutateTablet(assignment.tablet)
            .requireLocation(TabletMetadata.Location.future(assignment.server))
            .putLocation(TabletMetadata.Location.current(assignment.server))
            .deleteLocation(TabletMetadata.Location.future(assignment.server)).deleteSuspension();
...
        });
      }
```

Reply:
> Looking at this, is it possible to move the passing of the lock to the construction of the mutator?

I have no issue with that. I can make that change after all the tests are passing.

Reply:
I re-worked the service lock plumbing in 0243ff0.